nixos/modules/services/development/gemstash.nix

```diff
@@ -0,0 +1,158 @@
+  # on the nix side, so we rewrite plain names instead.
+  prefixColon = s: listToAttrs (map
+    (attrName: {
+      name = ":${attrName}";
+      value =
+        if isAttrs s.${attrName}
+        then prefixColon s."${attrName}"
+        else s."${attrName}";
+    })
+    (attrNames s));
+in
+{
+  options.services.gemstash = {
+    enable = mkEnableOption "gemstash service";
+    configFile = mkOption {
+      type = types.nullOr types.path;
+      default = null;
+      description = "Specify the path to a gemstash configuration file";
+    };
+
+    basePath = mkOption {
+      type = types.str;
+      default = null;
+      description = "Path to store the gem files and the sqlite database";
+    };
+    cacheType = mkOption {
+      type = types.nullOr (types.enum [ "memory" "memcached" ]);
+      default = null;
+      description = "How to cache values other than gem files (such as gem dependencies)";
+    };
+    memcachedServers = mkOption {
+      type = types.nullOr (types.listOf types.str);
+      default = null;
+      description = "The Memcached servers to connect to when using memcached";
+    };
+    dbAdapter = mkOption {
+      type = types.nullOr (types.enum [ "sqlite3" "postgres" "mysql" "mysql2" ]);
+      default = null;
+      description = "What database adapter to use";
+    };
+    dbUrl = mkOption {
+      type = types.nullOr types.str;
+      default = null;
+      description = "The database to connect to when using postgres, mysql, or mysql2";
+    };
+    dbConnectionOptions = mkOption {
+      type = types.nullOr types.attrs;
+      default = null;
+      description = "Additional Sequel.connect options to use";
+    };
+    rubyGemsUrl = mkOption {
+      type = types.nullOr types.str;
+      default = null;
+      description = "The default gem source URL";
+    };
+    ignoreGemfileSource = mkOption {
+      type = types.nullOr types.bool;
+      default = null;
+      description = "Ignore the source specified in Gemfile and always use :rubygems_url as gems upstream";
+    };
+    pumaThreads = mkOption {
+      type = types.nullOr types.int;
+      default = null;
+      description = "The number of threads used for the Gemstash server";
+    };
+    bind = mkOption {
+      type = types.nullOr types.str;
+      default = null;
+      description = "The binding used to start the Gemstash server";
+    };
+    protectedFetch = mkOption {
+      type = types.nullOr types.bool;
+      default = null;
+      description = "Tells Gemstash to authenticate via an API key before allowing the fetching of private gems and specs";
+    };
+    fetchTimeout = mkOption {
+      type = types.nullOr types.int;
+      default = null;
+      description = "The timeout setting for fetching gems";
+    };
+    logFile = mkOption {
+      type = types.nullOr types.str;
+      default = null;
+      description = "The name of the file to use for logging";
+    };
+
+    extraOptions = mkOption {
```

Comment:
Can be replaced by q single `settings` option. Please see my comment about RRC 42.